gobblin-dataset-config/src/main/java/gobblin/dataset/config/FileBasedConfigStore.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.dataset.config;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.log4j.Logger;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+
+/**
+ * This class is the implementation for the filed based config store. 
+ * @author mitu
+ *
+ */
+public class FileBasedConfigStore implements ConfigStore {
```

Comment:
Is this some prototype implementation or the "real deal"? What is the config layout? How are tags used? Has the config layout been discussed at the review and agreed on? Didn't we discuss that config is going to be deployed on HDFS? The current implementation seems to use regular files.